zerver/lib/actions.py

```diff
@@ -4479,15 +4479,16 @@ def do_change_bot_owner(
 ) -> None:
     previous_owner = user_profile.bot_owner
     user_profile.bot_owner = bot_owner
-    user_profile.save()  # Can't use update_fields because of how the foreign key works.
-    event_time = timezone_now()
-    RealmAuditLog.objects.create(
-        realm=user_profile.realm,
-        acting_user=acting_user,
-        modified_user=user_profile,
-        event_type=RealmAuditLog.USER_BOT_OWNER_CHANGED,
-        event_time=event_time,
-    )
+    with transaction.atomic():
+        user_profile.save()  # Can't use update_fields because of how the foreign key works.
```

Comment:
@timabbott Do you recall what specifically about "how the foreign key works" this is? This was added in 25ddba99f6eedf0bfdc7b9872fd06cec1b30da76 If not, I can investigate, but I think it'd be useful to improve this comment with that elaboration.

Reply:
I don't -- am going to try removing that line and seeing if at test fails.  https://github.com/zulip/zulip/pull/7458 seems to be the relevant PR.  @hackerkid FYI in case you remember this :)

Reply:
Yeah, `zerver.tests.test_events.NormalActionsTest.test_change_bot_owner` fails with this change.

Reply:
Sorry. But I can't seem to recollect anything regarding this :(